lms/djangoapps/class_dashboard/tests/test_dashboard_data.py

```diff
@@ -85,6 +98,36 @@ def setUp(self):
                     state=json.dumps({'attempts': self.attempts}),
                 )
 
+            # adding an instructor and a staff to the site. They should not be counted. Test should not fail
+            StudentModuleFactory.create(
+                grade=1,
+                max_grade=1,
+                student=instructor_member,
+                course_id=self.course.id,
+                module_state_key=self.item.location,
+                state=json.dumps({'attempts': self.attempts}),
+            )
+
+            StudentModuleFactory.create(
+                grade=1,
+                max_grade=1,
+                student=staff_member,
+                course_id=self.course.id,
+                module_state_key=self.item.location,
+                state=json.dumps({'attempts': self.attempts}),
+            )
+
+            # Adding this student will fail the test
+
+            # StudentModuleFactory.create(
+            #     grade=1,
+            #     max_grade=1,
+            #     student=newstudent,
+            #     course_id=self.course.id,
+            #     module_state_key=self.item.location,
+            #     state=json.dumps({'attempts': self.attempts}),
+            # )
+
             for j, user in enumerate(self.users):
                 StudentModuleFactory.create(
```

Comment:
Thanks @jbau. I fixed the existing code, and created StudentModules for the sequentials corresponding to the instructor and staff users. I also added the test cases per your suggestions and verified that there are 11 student StudentModules and 13 student+staff StudentModules, for both type ‘sequential’ and type ‘problem’. 
 
Now, the following test is failing and I think there is another bug.  It should be self.assertEquals(USER_COUNT, value), not self.assertEquals(0, value), right? 

def test_get_d3_sequential_open_distrib(self):
…...
......
         self.assertEquals(0, value)